File: python/histograma.py
def staircase(n):
    copy = n
    for i in range(n):
        cont = 0
        flag = 0
        while cont <= i:
            while flag < copy:
                print(' ',end='')
                flag+= 1
            print('#',end='')
            cont+= 1
        print()
        copy-= 1


n = int(input())
staircase(n)

File: C/histograma.c
#include <stdio.h>

int main()
{
    int qnt, i, cont, flag, copy;
    scanf("%d",&qnt);
    copy = qnt;
    for(i = 0; i < qnt; i++)
    {
        flag = 0;
        cont = 0;
        while(cont <= i)
        {
            while(flag < copy)
            {
                printf(" ");
                flag++;
            }
            printf("#");
            cont++;
        }
        printf("\n");
        copy--;
    }
}